cloudpickle_generators/_core.c

```diff
@@ -207,6 +207,9 @@ private_frame_data(PyObject* UNUSED(self), PyObject* frame_ob) {
 
     for (ix = 0; ix < size; ++ix) {
         PyObject* ob = frame->f_valuestack[ix];
+        if (ob == NULL) {
+            ob = Py_None;
```

Comment:
Instead of `Py_None`, what about using `&unset_value`? This is a sentinel that is used for unset closure cells which is fairly similar to this. The type of `&unset_value` can already be pickled. By using `None`, I am worried that code like:

```Python
f(None,(yield))
```
will fail because we will end up with something like:

```
LOAD_FAST('f')
LOAD_CONST(None)  # our explicit None
LOAD_CONST(None)  # the yielded value, implicitly None
YIELD_VALUE
CALL_FUNCTION(2)
```

If we serialize after the yield, we will have `None` on the stack, so if we set it to `NULL` when we deserialize, we will pass `NULL` to a Python function that expects `None`.

I am sort of surprised that they actually put `NULL` on the stack here during await; but w/e

Reply:
Sounds good. I’ll switch it over. A special sentinel is safer.